lib/ask-ai/ask-ai-with-previous-attempts.ts

```diff
@@ -6,21 +6,20 @@ interface AttemptHistory {
   error: string
 }
 
-export const askAiWithPreviousAttempts = async ({
-  prompt,
-  systemPrompt,
-  vfs,
-  previousAttempts,
-  onStream,
-  openaiClient,
-}: {
-  prompt: string
-  systemPrompt: string
-  vfs?: Record<string, string>
-  previousAttempts?: AttemptHistory[]
-  onStream?: (chunk: string) => void
-  openaiClient?: typeof openai
-}): Promise<string> => {
+export const askAiWithPreviousAttempts = async (
+  options: {
+    prompt: string
+    systemPrompt: string
+    previousAttempts?: AttemptHistory[]
+    onStream?: (chunk: string) => void
+  },
+  context: {
+    vfs?: Record<string, string>
+    openaiClient?: typeof openai
+  },
```

Comment:
ehh i would combine all of these options into a single object. It's not really clear what is context vs an option. The only thing that makes sense in context is the openaiClient i think